### DISCUSSION TITLE
Clicking and stuttering -- streams-generator-i2s

### DISCUSSION THREAD

**Discussion by nradina** [pschatzmann/arduino-audio-tools#1766] [Q&A] [answered]
Hi everyone,

Testing out a few DAC's today using a ESP32 and the PCM5102-DAC-Decoder and the HiLetgo PCM5102 I2S IIS Lossless Digital Audio DAC Decoder.

Success yet hearing a clicking and stuttering noise.
I've tried a few boards and also switched out the ESP32 with no change.

Attached is an example of the sound.

I'm sure this is my own error.
Thanks for any help!

https://github.com/user-attachments/assets/b84b2224-b257-43d5-9dda-018d7663a7e3

**Comment by pschatzmann** [comment, marked as answer]
You did not share what you are doing, so it is difficult to provide an answer: [double check this](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working).

My guess is: you must have some unnecessary delays in your audio loop!

**Reply by nradina** [reply]
Thank you, Phil.  I apologize for the limited information.

I am running the audio tools library example sketch: streams-generator-i2s.ino
No changes to the code.

Reading up now on your provided link.

**Comment by pschatzmann** [comment]
Did you change the log level to Warning ?
You can also lower your sampling rate...

**Reply by nradina** [reply]
Changing to "Warning" did the trick. Thank you, Phil.
